### ISSUE TITLE
Integration testing restricted routes

### ISSUE THREAD

**Issue by kingsloi** [httpoz/roles#44]
Hey, 

Great package! So simple and straight forward, it's been my go to for a while. 

I'm trying to write some integration tests so I can easily check that routes can/can't be accessed by people with/without a certain route, but I can't seem to get my tests to pass. 

I've asked the question on Stackoverflow and also Laracasts but no reply. I'm not sure if it's a bug within Laravel itself, or if I am incorrect in thinking that testing for this should be acceptance or something else.

Here's my test:

```
    public function testForbiddenRoleRoute()
    {
        $admin = factory(User::class)->create();
        $adminRole = \HttpOz\Roles\Models\Role::whereSlug('super.admin')->first();
        $admin->detachAllRoles();
        $admin->attachRole($adminRole);

        $this->actingAs($admin);

        $response = $this->json('GET', '/admin/users');

        $this->assertResponseStatus(200);
    }
```

Any ideas would be much appreciated!

**Comment by httpoz** [comment]
Thanks for using the package and I'm glad it's been working for you. So just off the top of my head, you have to create a factory for the roles as well. I am going to write a couple of tests for the package as well, so in the future I hope I will have done the grunt work for your use case.

**Comment by kingsloi** [comment]
Hey @httpoz, I actually have all the roles already seeded in my database. I have 3 roles, and 1 'group', no matter what combination I use I can't see to get it to work. 

I thought maybe it was Laravel's auth mixing something up, but it just looks the data isn't shared from my test -> middleware, so I'm not sure if it's a problem with the package, or with the Laravel core/tests. 

I can't seem to find much online either, so I'm assuming it's just me :(

**Comment by httpoz** [comment]
I'm going to start working on tests tomorrow and I will let you know when i have a solution for you, I hope that's not too late.

**Comment by kingsloi** [comment]
That'd be great! Very much appreciated!

**Comment by httpoz** [comment]
Sorry I took a while. I had the same problem, all my tests failed aswell. I'm not a hundred percent certain about the internals of Laravel with the method be and actingAs.

Take a look here: https://github.com/httpoz/roles/blob/develop/tests/features/RoleMiddlewareTest.php 

With the queried user I can verify a user has the role. However i can't with the created user and same challenge with actingAs. Might I suggest actually authenticating the user as opposed to using the actingAs method? I will try this myself tomorrow.

**Comment by kingsloi** [comment]
hey @httpoz, thank you for confirming the issue! I was really doubting myself. 

Do you think it's worth creating an issue on the Laravel core? I mean, it should work as we both expected, or maybe it doesn't and we're overlooking some limitation of `be()` or `actingAs()`

thanks for everything!

**Comment by httpoz** [comment]
I'd go with we are overlooking some limitation. In our applications, roles are assigned to an authenticated user, so I would suggest we authenticate/login the user first in our tests.

**Comment by kingsloi** [comment]
hey @httpoz, I figured it out... sorta. 

I posed my answer on Stackoverflow: https://stackoverflow.com/questions/44218439/laravel-5-4-integration-testing-role-middleware-fails/44331030#44331030

I ran into other issues if I use `$this->disableMiddleware()`, but I found a way to disable specific middleware using code I copied from a PR which is coming to L5.5, and disabled your middleware and it works great. 

I guess I have to move my role tests somewhere else! Acceptance maybe?
